app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt

```diff
@@ -117,10 +99,16 @@ class ScheduleActivity : ComponentActivity() {
 
         val futureDBSchedule: CompletableFuture<Schedule> = store.getSchedule(startMonday)
 
+        // todo move this into caching store
+        val locationProvider = (application as CoachMeApplication).locationProvider
+        locationProvider.updateContext(this, CompletableFuture.completedFuture(null))
+        val userLatLng = locationProvider.getLastLocation().value?: FusedLocationProvider.CAMPUS
+        val weatherState = store.getWeatherForecast(userLatLng).get()
```

Comment:
The get here is not ideal, although I am not sure how we could remove it. Using CompletableFuture<MutableState>> seems quite hard to deal with actually.

Reply:
Yup, again this follows your other comments, because for now `getWeatherForecast` returns a completedFuture of a mutable weather state.